solver/src/main.rs

```diff
@@ -192,3 +197,59 @@ async fn main() {
     serve_metrics(registry, ([0, 0, 0, 0], args.metrics_port).into());
     driver.run_forever().await;
 }
+
+async fn build_amm_artifacts(
+    sources: Vec<AmmSources>,
+    chain_id: u64,
+    settlement_contract: contracts::GPv2Settlement,
+    base_tokens: HashSet<H160>,
+    web3: web3::Web3<LoggingTransport<Http>>,
+) -> (PoolFetcher, Vec<UniswapLikeLiquidity>) {
```

Comment:
This method is a bit ugly at the moment because it is lopsided as the poolfetcher is constructed via the uniswap pair provider... Open to other suggestions of course.